src/main/java/cohen/weather/CurrentWeatherFrame.java

```diff
@@ -0,0 +1,94 @@
+package cohen.weather;
+
+import cohen.weather.jsonForFiveDay.FiveDayForecast;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+public class CurrentWeatherFrame extends JFrame
+{
+    private FiveDayForecast fiveDayForecast;
+
+    private final Retrofit retrofit = new Retrofit.Builder()
+            .baseUrl("https://api.openweathermap.org/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+            .build();
+
+    WeatherService service = retrofit.create(WeatherService.class);
+
+    String cityName = "New York";
+
+    CurrentWeatherView currentWeatherView;
+
+    public CurrentWeatherFrame()
+    {
+        fiveDayForecast = service.getFiveDayForecast(cityName).blockingFirst();
```

Comment:
blockingFirst() should only be used for tests.

Reply:
Not sure how to fix this because I run into issues with the subscribe() thread and the getList method

Reply:
What issues?

Reply:
If I remove blockingFirst, I need to change fiveDayForecast to Observable<FiveDayForecast> type, and then in CurrentWeatherView, I need to change the member field to also be that type. But then, this line doesn't work:
        List<InfoForFiveDay> info = fiveDayForecast.getList();
Because getList is a method of fiveDayForecast, not Observable<>. I tried adding the Disposable object like we did in the earthquakes project but I didn't do it properly, and I'm not if we should even be using it:

Disposable disposable = Observable
                .subscribeOn(Schedulers.io())
                .observeOn(Schedulers.newThread())
                .subscribe(
                        featureCollection -> {
                            List<InfoForFiveDay> info = fiveDayForecast.getList();
                        }
                        ,
                        Throwable::printStackTrace
                );

Reply:
That is how you're supposed to do it.